carmm/analyse/hirshfeld.py

```diff
@@ -0,0 +1,134 @@
+# Much recycled from mulliken.py, this should get Hirshfeld charges from aims.out
+
+def extract_hirshfeld(fname, natoms, data, write=True, outname='hirshfeld.txt'):
+    """
+
+    Args:
+
+        fname: Input file name, usually aims.out. str
+        natoms: number of atoms to extract charge from. int
+        data: type of output requested. str
+              Can be 'charge', 'volume', 'volume f', 'dipole vector', 'dipole moment' or 'second'
+        write: bool of whether to write the hirshfeld data to a new file called hirshfeld.txt
+
+    Returns: hirsh. List of requested data
+
+    """
+
+    # Extracts data from an aims.out and writes it to a new file.
+    import sys
+
+    ids = {
+        'charge':'Hirshfeld charge        :',
+        'volume':'Hirshfeld volume        :',
+        'volume f':'Free atom volume        :',
+        'dipole vector':'Hirshfeld dipole vector :',
+        'dipole moment':'Hirshfeld dipole moment :',
+        'second':'Hirshfeld second moments:'
+    }
+
+    if data in ids:
+        identifier = ids.get(data)
+
+    else:
+        print('Requested data not recognised')
+        sys.exit()
+
+        
+    with open(fname, 'r') as f:
+        output = f.readlines()
+
+    hirshfeld_line = get_hirshfeld_line(output)
+
+    if write is True:
+
+        write_hirshfeld(output, natoms, hirshfeld_line, outname)
+
+        hirsh = read_hirshfeld(outname, identifier)
+
+    else:
+        hirsh = read_hirshfeld(fname, identifier)
+
+    return hirsh
+
+
+def get_hirshfeld_line(text):
+
+    # Gets the line from which to read Hirshfeld data
+
+    hirshfeld_line = 0
+    output_line = len(text) - 1
+    while output_line and not hirshfeld_line:
+        if "Performing Hirshfeld analysis of fragment charges and moments." in text[output_line]:
+            hirshfeld_line = output_line
+
+        else:
+            output_line -= 1
+
+    return hirshfeld_line
+
+
+def write_hirshfeld(text, natoms, start, outname):
+
```

Comment:
It would be nice if you could write a docstring for this function. I got confused about the 'start' argument.

Reply:
Its an internal function that shouldn't really be seen by a user, but I agree that it is currently opaque